source/server/server.cc

```diff
@@ -138,9 +146,12 @@ void InstanceImpl::flushStats() {
   server_stats_->total_connections_.set(numConnections() + info.num_connections_);
   server_stats_->days_until_first_cert_expiring_.set(
       sslContextManager().daysUntilFirstCertExpires());
-
-  InstanceUtil::flushCountersAndGaugesToSinks(config_->statsSinks(), stats_store_);
-  stat_flush_timer_->enableTimer(config_->statsFlushInterval());
+  // TODO(ramaraochavali): consider adding different flush interval for histograms.
+  if (!terminated_) {
+    stats_store_.mergeHistograms();
```

Comment:
Same comment applies, this is going to be one cycle behind since this will not block and the post back will come after this runs. If we don't care about this, please add some comments to this effect. I still think megeHistograms() should take a lambda which is invoked when the merge is complete.

Reply:
I agree that sending metrics one cycle behind is not good. I have added the lambda to mergeHistograms which will complete the final stage of stats flush